src/change_stream.ts

```diff
@@ -302,62 +313,29 @@ export class ChangeStream extends EventEmitter {
 
       // Tidy up the existing cursor
       const cursor = this.cursor;
+      const cursorStream = cursor[kCursorStream];
+      if (!cursorStream) throw new MongoError('No stream on cursor');
```

Comment:
Similar to above, I'm very surprised this isn't being triggered anywhere in the tests. This is a very aggressive check that I would expect to throw in cases where we only use the `ChangeStream` as an iterator, not as an event emitter.